src/java/org/apache/cassandra/db/rows/BTreeRow.java

```diff
@@ -399,9 +399,9 @@ public boolean hasComplex()
 
     public boolean hasComplexDeletion()
     {
-        long result = accumulate((cd, v) -> ((ComplexColumnData) cd).complexDeletion().isLive() ? 0 : Cell.MAX_DELETION_TIME,
+        long result = accumulate((cd, v) -> ((ComplexColumnData) cd).complexDeletion().isLive() ? 0 : Long.MAX_VALUE,
                                  COLUMN_COMPARATOR, isStatic() ? FIRST_COMPLEX_STATIC : FIRST_COMPLEX_REGULAR, 0L);
-        return result == Cell.MAX_DELETION_TIME;
+        return result == Long.MAX_VALUE;
```

Comment:
I simply reverted here rather than trying to do something less brittle around the stop value. Open to doing that if we think it really helps...

Reply:
So when working on 14227 the whole code was littered with magical sentinel values and it was very difficult to make sense of things. I did my best to provide semantic meaning to those with proper constant names and apparently broke this one. Could you name this sthg meaningful?